bluesky/run_engine.py

```diff
@@ -305,6 +311,60 @@ def __init__(self, md=None, *, loop=None, preprocessors=None,
 
         self.loop.call_soon(self._check_for_signals)
 
+    @property
+    def commands(self):
+        '''
+        The list of commands available to Msg.
+
+        See Also
+        --------
+        :meth:`RunEngine.register_command`
+        :meth:`RunEngine.unregister_command`
+        :meth:`RunEngine.print_command_registry`
+
+        Examples
+        --------
+        >>> from bluesky import RunEngine
+        >>> RE = RunEngine()
+        >>> # to list commands
+        >>> RE.commands
+        '''
+        # return as a list, not lazy loader, no surprises...
+        return list(self._command_registry.keys())
+
+    def print_command_registry(self, verbose=False):
+        '''
+        This conveniently prints the command registry of available
+        commands.
+
+        Parameters
+        ----------
+        Verbose : bool, optional
+        verbose print. Default is False
+
+        See Also
+        --------
+        :meth:`RunEngine.register_command`
+        :meth:`RunEngine.unregister_command`
+        :attr:`RunEngine.commands`
+
+        Examples
+        --------
+        >>> from bluesky import RunEngine
+        >>> RE = RunEngine()
+        >>> # Print a very verbose list of currently registered commands
+        >>> RE.print_command_registry(verbose=True)
+        '''
+        commands = "List of available commands\n"
+
+        for command, func in self._command_registry.items():
+            docstring = func.__doc__
+            if verbose is False:
```

Comment:
`if not verbose` is a bit more Pythonic.

Reply:
thanks! done